distribution/lib/Standard/Base/0.0.0-dev/src/Data/Json/Extensions.enso

```diff
@@ -1,136 +1,137 @@
+   For Nothing, booleans, numbers and strings, this is the value itself.
+   For arrays or vectors, the elements are converted recursively.
+   For atoms and maps, the object is converted to a JS_Object.
+Text.to_js_object : JS_Object | Text
+Text.to_js_object self = case self of
+    Text -> JS_Object.from_pairs [["type", "Text"]]
+    _ -> self
+
+## Converts the given value to a JSON serializable object.
+   For Nothing, booleans, numbers and strings, this is the value itself.
+   For arrays or vectors, the elements are converted recursively.
+   For atoms and maps, the object is converted to a JS_Object.
+Number.to_js_object : JS_Object | Number
+Number.to_js_object self = case self of
+    Number -> JS_Object.from_pairs [["type", "Number"]]
+    Integer -> JS_Object.from_pairs [["type", "Integer"]]
+    Decimal -> JS_Object.from_pairs [["type", "Decimal"]]
+    _ -> self
+
+## Converts the given value to a JSON serializable object.
+   For Nothing, booleans, numbers and strings, this is the value itself.
+   For arrays or vectors, the elements are converted recursively.
+   For atoms and maps, the object is converted to a JS_Object.
+Boolean.to_js_object : JS_Object | Boolean
+Boolean.to_js_object self = case self of
+    Boolean -> JS_Object.from_pairs [["type", "Boolean"]]
+    _ -> self
+
+## Converts the given value to a JSON serializable object.
+   For Nothing, booleans, numbers and strings, this is the value itself.
+   For arrays or vectors, the elements are converted recursively.
+   For atoms and maps, the object is converted to a JS_Object.
+Nothing.to_js_object : Nothing
+Nothing.to_js_object self = self
+
+## Converts the given value to a JSON serializable object.
+   For Nothing, booleans, numbers and strings, this is the value itself.
+   For arrays or vectors, the elements are converted recursively.
+   For atoms and maps, the object is converted to a JS_Object.
+Array.to_js_object : Vector
+Array.to_js_object self =
+    proxy = Array_Proxy.new self.length i-> self.at i . to_js_object
+    Vector.from_polyglot_array proxy
+
+## Converts the given value to a JSON serializable object.
+   For Nothing, booleans, numbers and strings, this is the value itself.
+   For arrays or vectors, the elements are converted recursively.
+   For atoms, the object is converted to a JS_Object.
+Vector.to_js_object : Vector
+Vector.to_js_object self =
+    proxy = Array_Proxy.new self.length i-> self.at i . to_js_object
+    Vector.from_polyglot_array proxy
+
+## Converts the given value to a JSON serializable object.
+   For Nothing, booleans, numbers and strings, this is the value itself.
+   For arrays or vectors, the elements are converted recursively.
+   For atoms, the object is converted to a JS_Object.
+Any.to_js_object : JS_Object
+Any.to_js_object self =
     m = Meta.meta self
     case m of
         _ : Meta.Atom ->
+            type_pair = ["type", Meta.get_qualified_type_name self . split '.' . last]
```

Comment:
I'm wondering if
```suggestion
            type_pair = ["type", Meta.type_of self . to_text]
```

won't be simpler, while it should provide the same result, if I'm not mistaken.